### ISSUE TITLE
Error provisioning  the container app. Container 'XXX' failed to start-up. Please check Log Analytics workspace for container logs

### ISSUE THREAD

**Issue by andreatosato** [microsoft/azure-container-apps#56]
Hi, 
I have some errors to deploy DAPR application on Container apps.
This is my script: https://github.com/andreatosato/dapr-cloudchampion/tree/azure-container-apps/deploy/AzureContainerApps
This is console logs:

```
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:693: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:697: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if containerapp_def.template.containers[0].env is None or len(containerapp_def.template.containers[0].env) is 0:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:126: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:137: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(value) is 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:153: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:174: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
Command group 'containerapp' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Deployment failed. Correlation ID: a78614e0-2dee-4a35-8049-186b8c5ef93c. Failed to provision revision for container app 'web'. Error details: Code: 90714, ExtendedCode: 90714, Message: Error provisioning the container app. Container 'web' failed to start-up. Please check Log Analytics workspace for container logs..
Command ran in 42.618 seconds (init: 0.229, invoke: 42.389)
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:693: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:697: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if containerapp_def.template.containers[0].env is None or len(containerapp_def.template.containers[0].env) is 0:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:126: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:137: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(value) is 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:153: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:174: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
Command group 'containerapp' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Deployment failed. Correlation ID: 2deded40-53f0-4ec5-8cf3-26022669ee2b. Failed to provision revision for container app 'order'. Error details: Code: 90714, ExtendedCode: 90714, Message: Error provisioning 
the container app. Container 'order' failed to start-up. Please check Log Analytics workspace for container logs..
Command ran in 37.587 seconds (init: 0.219, invoke: 37.368)
```

**Comment by andreatosato** [comment]
[query_data.csv](https://github.com/microsoft/azure-container-apps/files/7683470/query_data.csv)

This is LogAnalitycs errors

**Comment by kendallroden** [comment]
@andreatosato Hi Andrea. I can't see any output from the logs based on the above csv file. All of the Log_s entries are blank on my end. Can you confirm you had no log output or if there may be some data loss. 

Also- I am having trouble getting the solution running on my local machine using the docker compose files. I am getting a warning when I run the order service locally that says there was a failure to determine the https port for redirection

**Comment by kendallroden** [comment]
I am currently testing it myself and if I am able to repro we will dig into this on the backend

**Comment by andreatosato** [comment]
@kendallroden now I'm not in front of PC.
I have data on log field.
I saw only log information of from dapr

**Comment by andreatosato** [comment]
This is only configuration that I need to change in my app at runtime.

https://github.com/andreatosato/dapr-cloudchampion/blob/dad512ffa5ff8c1a5e96f8184de14fc65a404ae9/Frontend/CloudChampion.Web/Pages/Index.razor#L73

Can I use secrets (azure container app secrets) for my purpose?



Each service run in http.

I have frontend "blazor" that call service.

Service use "state management" and "pubsub"

**Comment by andreatosato** [comment]
@kendallroden must I run app on https?

**Comment by kendallroden** [comment]
No http is fine. I am testing now. I will get back to you @andreatosato

**Comment by kendallroden** [comment]
I put in a pull request to fix your deployment manifests. I also added a comment around pub sub connection string- it should be the root sb connection string, not a topic-specific connection string. The issue I ran into after they were deployed is in your connection to the backend service from the blazor app. I assume it's because of how you are currently routing. Because it's a SPA, are you using any type of server in the blazor code to act as a proxy when you connect to the order service? I am assuming the answer is no. In this [code repo](www.github.com/reddog-containerapps) you can see how we deployed traefik and potentially use this to aid in your routing if needed. Right now all the apps in a CA environment share a domain, so depending on your implementation you may need to consider this. I will dig in a bit more to see if I can get it working, but curious on your thoughts!

**Comment by kendallroden** [comment]
I just thought of something. You have ingress complete disabled for the order service right now! I will test out making that internally accessible as well

**Comment by kendallroden** [comment]
@andreatosato is your signalr implementation based on tcp? @andreatosato

**Comment by andreatosato** [comment]
Signalr Is based on websocket

**Comment by andreatosato** [comment]
Take a look to connectionstring: https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-servicebus/

**Comment by kendallroden** [comment]
Okay! Perhaps that is not necessary. Spent most of today digging around on this. You should be unblocked getting the deployment into container apps. If the application can communicate over http or https you are good to go, however you will need to consider how you will dynamically configure the url, so checking out the samples above might help. Let me know if you are able to work through it- it's 6:30 pm my time on a Friday so I am going to stop working, but happy to support further if needed. @andreatosato

**Comment by andreatosato** [comment]
I use blazor-server webapp, so, I have clientside (c# page) that must connect to Signalr hub (server side).
This connection is intra-app so we don't root conversation outside container or to dapr sidecar container.


I have 2 flow:
1 flow (service to service invocation):
Blazor client -> dapr --- dapr-> service

2 flow
Blazor client -> dapr --- dapr -> service -> dapr --- dapr -> pubsub --- dapr -> hub -> websocket -> blazor client

**Comment by andreatosato** [comment]
Tomorrow I see your link and try something.

1:30 am here 🤗
Thanks for your great work

**Comment by kendallroden** [comment]
Thanks for clarifying. TBH I wrote an app with Signal R and Vue a couple of years ago and I am rusty. I see now how the wiring up happens in program.cs to map the NotificationHub to an endpoint /notification. I'll keep digging around bc now I am invested :) Would love to see this running in Capps.

**Comment by andreatosato** [comment]
I have 2 community session next week and 1 in January.
I want to use azure container apps as main topic in next 6 months.

I have many repo to deploy on this new service:

https://github.com/andreatosato/SpotifyMicroservices

And this
https://github.com/andreatosato/Codemotion-2021-11/blob/main/docs/img/OpenTelemetry-Microservices.png

**Comment by kendallroden** [comment]
Okay slight progress. I got the UI to appear without an error by setting the url to 127.0.0.1 instead of the service name. Now just have to figure out the error when I submitted an order

**Comment by andreatosato** [comment]
Oh, great!
So, now error is related to dapr communication.

**Comment by andreatosato** [comment]
[query_data.xlsx](https://github.com/microsoft/azure-container-apps/files/7696621/query_data.xlsx)
I upgrade my deployments with your pull request but result is the same.
I attacch LogAnalitycs.

Can I see container log? Can I see dapr log?

**Comment by andreatosato** [comment]
I redeploy all resource:
This is my error:
```
time="2021-12-11T02:21:03.289900756Z" level=warning msg="Failed to listen on :50002 with error: listen tcp :50002: bind: address already in use" app_id=cloudchampion-order instance=order--eb9d2z8-6979d87946-kg5rc scope=dapr.runtime.grpc.internal type=log ver=edge
```
[query_data (2).xlsx](https://github.com/microsoft/azure-container-apps/files/7696665/query_data.2.xlsx)

**Comment by andreatosato** [comment]
@kendallroden 
I update demo for local environment: [here](https://github.com/andreatosato/dapr-cloudchampion/commit/35c29e1a309074f94fd5e82147c225d2fa55b0af)

Now Azure Service Bus works. I configure 2 topic. 
![image](https://user-images.githubusercontent.com/24985956/145673850-817f13a1-05e9-461b-bef7-7bdb3bc7df08.png)

I change connections string and I remove EntityPath.


Azure Container apps has same error:
`time="2021-12-11T02:21:03.289900756Z" level=warning msg="Failed to listen on :50002 with error: listen tcp :50002: bind: address already in use" app_id=cloudchampion-order instance=order--eb9d2z8-6979d87946-kg5rc scope=dapr.runtime.grpc.internal type=log ver=edge`

**Comment by jeffhollan** [comment]
Just cloned the sample and created in a new container app environment. I hit one more bug, but it did work end to end. @andreatosato maybe try deleting the container and recreating or deleting the environment? I wonder if it's in a bad state right now.

Btw the last change I had to make in your sample is in your container app deploy script you call it the `state-management` state component, but in your code you have a typo and try to call the `state-managment` (missing an E). So you either need to mirror the typo in the component, or fix the code where you have this typo in a few spots

https://github.com/andreatosato/dapr-cloudchampion/blob/35c29e1a309074f94fd5e82147c225d2fa55b0af/deploy/AzureContainerApps/components.yaml#L7

Cool demo. @kendallroden glad you were able to find the 127.0.0.1 piece. That saved a ton of time

![image](https://user-images.githubusercontent.com/7132626/145682786-95fea690-8842-422d-88f3-fdacaa8750da.png)

**Comment by andreatosato** [comment]
I delete enviroment many time.
I fix typo and I push container again. I recreate containerapp, log analitycs and apps.
https://github.com/andreatosato/dapr-cloudchampion/commit/db9fcf621714526ab440cf1e7ca4fd08fc76f83b

But demo don't not works.

`time="2021-12-11T02:21:03.289900756Z" level=warning msg="Failed to listen on :50002 with error: listen tcp :50002: bind: address already in use" app_id=cloudchampion-order instance=order--eb9d2z8-6979d87946-kg5rc scope=dapr.runtime.grpc.internal type=log ver=edge`

**Comment by jeffhollan** [comment]
Where do you see that last error? Is it what's returned from the az create command? Or in log analytics?

**Comment by andreatosato** [comment]
Log analitycs

**Comment by andreatosato** [comment]
Commands result
```
This command is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
You already have the latest azure-cli version: 2.31.0
Upgrading extensions
Checking update for connectedk8s
No updates available for 'connectedk8s'. Use --debug for more information.
Checking update for containerapp
No updates available for 'containerapp'. Use --debug for more information.
Upgrade finished.You can enable auto-upgrade with 'az config set auto-upgrade.enable=yes'. More details in https://docs.microsoft.com/cli/azure/update-azure-cli#automatic-update
{
  "createdDate": "Wed, 08 Dec 2021 09:27:18 GMT",
  "customerId": "aed8598b-4e5a-4b09-a3ae-328c0fa1011f",
  "eTag": null,
  "features": {
    "clusterResourceId": null,
    "disableLocalAuth": null,
    "enableDataExport": null,
    "enableLogAccessUsingOnlyResourcePermissions": true,
    "immediatePurgeDataOn30Days": null,
    "legacy": 0,
    "searchVersion": 1
  },
  "forceCmkForQuery": null,
  "id": "/subscriptions/eab27a5f-08dd-4e9d-bd26-21caf854af94/resourcegroups/containerapps-tosato/providers/microsoft.operationalinsights/workspaces/workspace-logs",
  "location": "canadacentral",
  "modifiedDate": "Sat, 11 Dec 2021 15:56:28 GMT",
  "name": "workspace-logs",
  "privateLinkScopedResources": null,
  "provisioningState": "Succeeded",
  "publicNetworkAccessForIngestion": "Enabled",
  "publicNetworkAccessForQuery": "Enabled",
  "resourceGroup": "containerapps-tosato",
  "retentionInDays": 30,
  "sku": {
    "capacityReservationLevel": null,
    "lastSkuUpdate": "Wed, 08 Dec 2021 09:27:18 GMT",
    "name": "pergb2018"
  },
  "tags": null,
  "type": "Microsoft.OperationalInsights/workspaces",
  "workspaceCapping": {
    "dailyQuotaGb": -1.0,
    "dataIngestionStatus": "RespectQuota",
    "quotaNextResetTime": "Sun, 12 Dec 2021 00:00:00 GMT"
  }
}
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:693: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:697: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if containerapp_def.template.containers[0].env is None or len(containerapp_def.template.containers[0].env) is 0:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:126: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:137: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(value) is 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:153: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:174: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
Command group 'containerapp env' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
{
  "aksResourceId": null,
  "appLogsConfiguration": {
    "destination": "log-analytics",
    "logAnalyticsConfiguration": {
      "customerId": "aed8598b-4e5a-4b09-a3ae-328c0fa1011f",
      "sharedKey": null
    }
  },
  "arcConfiguration": null,
  "containerAppsConfiguration": {
    "aciSubnetResourceName": null,
    "appSubnetResourceId": null,
    "controlPlaneSubnetResourceId": null,
    "daprAIInstrumentationKey": null,
    "dockerBridgeCidr": null,
    "platformReservedCidr": null,
    "platformReservedDnsIP": null,
    "subnetResourceId": null
  },
  "defaultDomain": "politerock-708269b0.canadacentral.azurecontainerapps.io",
  "deploymentErrors": null,
  "extendedLocation": null,
  "id": "/subscriptions/eab27a5f-08dd-4e9d-bd26-21caf854af94/resourceGroups/ContainerApps-Tosato/providers/Microsoft.Web/kubeEnvironments/development",
  "internalLoadBalancerEnabled": null,
  "kind": "containerenvironment",
  "kubeEnvironmentType": "managed",
  "location": "canadacentral",
  "name": "development",
  "provisioningState": "Succeeded",
  "resourceGroup": "ContainerApps-Tosato",
  "staticIp": "20.151.106.163",
  "tags": null,
  "type": "Microsoft.Web/kubeenvironments"
}
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:693: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:697: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if containerapp_def.template.containers[0].env is None or len(containerapp_def.template.containers[0].env) is 0:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:126: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:137: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(value) is 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:153: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:174: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
Command group 'containerapp' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Deployment failed. Correlation ID: b6ee9c56-a825-485b-8abe-3b939370895e. Failed to provision revision for container app 'web'. Error details: Code: 90714, ExtendedCode: 90714, Message: Error provisioning the container app. Container 'web' failed to start-up. Please check Log Analytics workspace for container logs..
Command ran in 44.696 seconds (init: 0.206, invoke: 44.490)
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:693: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\custom.py:697: SyntaxWarning: "is" with a literal. Did you mean "=="?    
  if containerapp_def.template.containers[0].env is None or len(containerapp_def.template.containers[0].env) is 0:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:126: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:137: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(value) is 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:153: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
C:\Users\Andrea Tosato\.azure\cliextensions\containerapp\azext_workerapp\_utils.py:174: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if len(key_val) is not 2:
Command group 'containerapp' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Deployment failed. Correlation ID: 7ee7d4ee-27ec-4abe-a779-d4a853ea03c1. Failed to provision revision for container app 'order'. Error details: Code: 90714, ExtendedCode: 90714, Message: Error provisioning the container app. Container 'order' failed to start-up. Please check Log Analytics workspace for container logs..
Command ran in 45.661 seconds (init: 0.205, invoke: 45.456)
```

**Comment by andreatosato** [comment]
![image](https://user-images.githubusercontent.com/24985956/145683956-3fc1f933-aefe-48f0-bd45-e1d1de343d41.png)

**Comment by jeffhollan** [comment]
Hmm I wonder if there's another issue with these container images starting up and dapr failing is just one side effect of a visible err. can you try with the container images I just used in my ACR? You should be able to pull these without access. If it works you may want to rebuild and republish your images:

jeffhollan.azurecr.io/web:test
jeffhollan.azurecr.io/order:test

**Comment by andreatosato** [comment]
Oh oh oh!
@jeffhollan works!

https://web.wittymushroom-045b656a.canadacentral.azurecontainerapps.io/

Great!

How do you create container? 
docker-compose o docker build of each app?

**Comment by jeffhollan** [comment]
Docker build each app. But I'd expect compose should work too but not how I did it

**Comment by andreatosato** [comment]
I need to investigate on docker build.
I close issue and try again.

Thanks a lot @jeffhollan @kendallroden

**Comment by jeffhollan** [comment]
Actually specifically I ran az acr build and had ACR build and publish

**Comment by andreatosato** [comment]
az acr build --image web:az -r containerappstosato.azurecr.io .\Frontend\CloudChampion.Web\

No build for me

**Comment by jeffhollan** [comment]
`az acr build -r jeffhollan -t web:az -f Frontend/CloudChampion.Web/Dockerfile .`
